pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -164,9 +165,13 @@ def __init__(
             self._set_training_type_plugin()
         else:
             self.set_distributed_mode()
-        self._configure_slurm_ddp()
 
         self.handle_given_plugins()
+        self._set_distrib_type_if_training_type_plugin_passed()
+
+        self._configure_slurm_ddp()
+        self.select_cluster_environment()
```

Comment:
This function does not set state but instead returns an object, what does this line do then?

Reply:
Resolved!